drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2077 @@
+ */
+
+const char *const gpio_pca_series_reg_name[] = {
+	"1b_input_port",
+	"1b_output_port",
+/*  "1b_polarity_inversion," */
+	"1b_configuration",
+	"2b_output_drive_strength",
+	"1b_input_latch",
+	"1b_pull_enable",
+	"1b_pull_select",
+	"1b_input_status",
+	"1b_output_config",
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	"1b_interrupt_mask",
+	"1b_interrupt_status",
+	"2b_interrupt_edge",
+	"1b_interrupt_clear",
+#endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */
+	"reg_end",
+};
+
+/**
+ * @brief drive strength for output_drive_strength register
+ *
+ * @note Only applies to part no with PCA_HAS_LATCH capability.
+ */
+enum gpio_pca_series_drive_strength {
+	gpio_pca_series_strength_x1 = 0U, /** Lowest strength */
+	gpio_pca_series_strength_x2,
+	gpio_pca_series_strength_x3,
+	gpio_pca_series_strength_x4, /** Highest strength, default */
+};
+
+/**
+ * @brief interrupt config for interrupt_edge register
+ *
+ * @note Only applies to part no with PCA_HAS_INT_EXTEND capability.
+ */
+enum gpio_pca_series_interrupt_config_extended {
+	gpio_pca_series_interrupt_level_change = 0U, /* default */
+	gpio_pca_series_interrupt_rising_edge,
+	gpio_pca_series_interrupt_falling_edge,
+	gpio_pca_series_interrupt_either_edge,
+};
+
+struct gpio_pca_series_part_config {
+	enum gpio_pca_series_part_no part_no; /* enum of part unmber */
+	uint8_t port_no;                      /* number of 8-pin ports on device */
+	uint8_t flags;                        /* capability flags */
+	const uint8_t *regs;                  /* pointer to register map */
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+	const uint8_t *cache_map;
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+};
+
+/** Configuration data */
+struct gpio_pca_series_config {
+	struct gpio_driver_config common; /* gpio_driver_config needs to be first */
+	struct i2c_dt_spec i2c;           /* i2c bus dt spec */
+	const struct gpio_pca_series_part_config *part_cfg; /* config of part unmber */
+	struct gpio_dt_spec gpio_rst;                       /* device reset gpio */
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	struct gpio_dt_spec gpio_int; /** device interrupt gpio */
+#endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */
+};
+
+/** Runtime driver data */
+struct gpio_pca_series_data {
+	struct gpio_driver_data common; /** gpio_driver_data needs to be first */
+	struct k_sem lock;
+	void *cache;  /** device spicific reg cache
+			*  - if CONFIG_GPIO_PCA_SERIES_CACHE_ALL is set,
+			*    it points to device specific cache memory.
+			*  - if CONFIG_GPIO_PCA_SERIES_CACHE_ALL is not set,
+			*    it point to a struct gpio_pca_series_reg_cache_mini
+			*    instance.
+			*/
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	const struct device *self;    /** Self-reference to the driver instance */
+	struct gpio_callback gpio_cb; /** gpio_int ISR callback */
+	sys_slist_t callbacks;        /** port pin callbacks list */
+	struct k_work int_work;       /** worker that fire callbacks */
+#endif
+};
+
+/*!
```

Comment:
I'm not clear why doxygen style macros are being used here- I have no issue with doxygen style function documentation in drivers, but these kinds of macros are only really useful when doxygen will be parsing the file- and to my knowledge doxygen does not parse the driver files themselves- so creating a doxygen group here won't be useful

Reply:
Yeah, it is just used to remind me what these code does. We can remove these if confusing.

Reply:
Maybe instead we expand the comment? We aren't limited by doxygen syntax here, this commend could instead say "PCA GPIO register access functions"

Reply:
changed this to avoid creating doxygen groups. it seems they are causing ci doc build failure.